sync/task_pool.go

```diff
@@ -249,10 +249,20 @@ func (p *TaskPool) IsClosed() bool {
 	}
 }
 
+// Close close, terminate task
 func (p *TaskPool) Close() {
 	p.stop()
 	p.wg.Wait()
 	for i := range p.qArray {
 		close(p.qArray[i])
 	}
 }
+
+// CloseTillTaskComplete close till task complete
+func (p *TaskPool) CloseTillTaskComplete() {
```

Comment:
I think u can add a param for Close() to decide the exec sequence of stop and wait instead of  add a new func 'CloseTillTaskComplete'.